pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -2508,34 +2512,30 @@ private String receiveCommandStatus() throws IOException {
   }
 
   private void interpretCommandStatus(String status, ResultHandler handler) {
-    int update_count = 0;
-    long insert_oid = 0;
-
-    if (status.startsWith("INSERT") || status.startsWith("UPDATE") || status.startsWith("DELETE")
-        || status.startsWith("MOVE")) {
+    long oid = 0;
+    int count = 0;
+    Matcher matcher = COMMAND_COMPLETE_PATTERN.matcher(status);
```

Comment:
What do you think of reusing `Matcher` instance?
Matcher could be cached in a `QueryExecutorImpl`.

Reply:
`Matcher` isn't thread safe[1] so it's cleaner to leave it internal to the the method. The main work of prepping the regex is done by the `Pattern.compile(...)` anyway so I doubt a cached Matcher would make much difference. The standard approach of regex parsing in Java is to have a static `Pattern` and a local variable for the matcher as it handles both the matching and groups for the results.

[1]: *... and yes neither is the Connection or QueryExecutor but you know what I mean...*

Reply:
`create table update_bench_test(int_field int4, str_field varchar(200))`
`connection.prepareStatement("update update_bench_test set int_field = int_field+1")` (the table is empty)

No matcher reuse: the code allocates 624 bytes. With matcher reuse -- 376 bytes.

Reply:
Sure but the matcher would be young gen allocation and be easily reclaimed.

I think pulling it outside the method comes at the expense of readability for negligible gain. If you think it's worth it I'd suggest doing it in a follow up PR.